pipeline_dash/viz/dash/components/jobs_pipeline_fig.py

```diff
@@ -59,74 +72,100 @@ def generate_plot_figure(graph: networkx.DiGraph) -> go.Figure:
             autosize=True,
-            height=y_scale * 15,
             hovermode="closest",
             margin=go.layout.Margin(b=0, t=0, l=0, r=0),
-            meta=dict(
-                default_edge_width=default_edge_width,
-                default_node_size=default_node_size,
-                default_scaling=default_scaling,
-                default_yaxis_range=[-4, y_scale],
-            ),
             selectdirection="v",
             showlegend=False,
             titlefont_size=16,  # type: ignore
-            uirevision=str(uuid.uuid4()),
-            updatemenus=layout_buttons,
             xaxis=dict(
                 showgrid=False,
                 showticklabels=False,
                 zeroline=False,
             ),
             yaxis=dict(
                 constraintoward="top",
-                range=[-4, y_scale],
                 showgrid=False,
                 showticklabels=False,
                 zeroline=False,
             ),
         ),
     )
-    # fig.update_layout(annotations=annotations)
+    fig.update(
+        data=(*edge_traces.values(), node_trace),
+        overwrite=True,
+    )
+    fig.update_layout(
+        height=y_scale * 15,
+        meta=dict(
+            default_edge_width=default_edge_width,
+            default_node_size=default_node_size,
+            default_scaling=default_scaling,
+            default_yaxis_range=[-4, y_scale],
+        ),
+        uirevision=str(uuid.uuid4()),
+        overwrite=True,
+    )
+    fig.update_yaxes(range=[-4, y_scale], overwrite=True)
+    show_annotations = bool(y_scale < y_scale_limit)
+    cache.set(f"{session_id}.graph", graph)
 
     end_time = time.process_time()
     print(f"Rendered graph in {end_time - start_time} sec")
-    return fig
+    return fig, show_annotations
 
 
-def get_node_labels(graph, node_text_dict):
-    annotations = [
+LayoutUpdate = dict
+
+
+# @pcprofile
+def generate_annotations_layout_update(
+    graph: networkx.DiGraph, session_id: str, show_annotations: bool
+) -> tuple[LayoutUpdate, tuple[go.Annotation, ...]]:
+    annotations = get_node_labels(graph) if show_annotations else tuple()
+    cache.set(f"{session_id}.annotations", annotations)
+    default_scaling = cache.get(f"{session_id}.figure_default_scaling")
+    size_annotations(default_scaling, annotations)
+    # layout_buttons = list(
+    #     [
+    #         go.layout.Updatemenu(
+    #             type="dropdown",
+    #             active=0 if show_annotations else 1,
+    #             buttons=list(
+    #                 [
+    #                     dict(label="Label:On", method="relayout", args=[{"annotations": annotations}]),
+    #                     dict(label="Label:Off", method="relayout", args=[{"annotations": []}]),
+    #                 ]
+    #             ),
+    #             showactive=False,
+    #             pad=dict(b=0, l=0, r=0, t=0),
+    #             x=0.05,
+    #             y=1,
+    #         )
+    #     ]
+    # )
```

Comment:
dead code?

Reply:
Yea, holdover from when I tried to make plotly in-figure menus work. But those menus are utter garbage